returns/_generated/pipeline/managed.py

```diff
@@ -1,125 +1,10 @@
-
-    Implementation
-    ~~~~~~~~~~~~~~
-    This class requires some explanation.
-
-    First of all, we modeled this function as a class,
-    so it can be partially applied easily.
-
-    Secondly, we used imperative approach of programming inside this class.
-    Functional approached was 2 times slower.
-    And way more complex to read and understand.
-
-    Lastly, we try to hide these two things for the end user.
-    We pretend that this is not a class, but a function.
-    We also do not break a functional abstraction for the end user.
-    It is just an implementation detail.
-
-    Type inference does not work so well with ``lambda`` functions.
-    But, we do not recommend to use this function with ``lambda`` functions.
-
-    """
-
-    __slots__ = ('_use', '_release')
-
-    def __init__(self, use, release):
-        """Saving the ``use`` and ``release`` steps."""
-        object.__setattr__(self, '_use', use)  # noqa: WPS609
-        object.__setattr__(self, '_release', release)  # noqa: WPS609
-
-    def __call__(self, acquire):
-        """
-        Calling the pipeline by providing the first ``acquire`` step.
-
-        It might look like a typeclass,
-        but typeclass support is not yet enabled in our project.
-        So, it is just a bunch of ``if`` statements for now.
-        """
-        if isinstance(acquire, IOResult):
-            return acquire.bind(self._ioresult_pipeline)
-        elif isinstance(acquire, RequiresContextIOResult):
-            return acquire.bind(self._reader_ioresult_pipeline)
-        elif isinstance(acquire, RequiresContextFutureResult):
-            return acquire.bind(self._reader_future_result_pipeline)
-        return acquire.bind_async(self._future_pipeline)
-
-    def _ioresult_pipeline(self, inner_value):
-        used = self._use(inner_value)
-        inner_result = used._inner_value  # noqa: WPS437
-        self._release(inner_value, inner_result)
-        return used
-
-    def _reader_ioresult_pipeline(self, inner_value):
-        def factory(deps):
-            return _managed(
-                lambda inner: self._use(inner)(deps),
-                lambda inner, pure: self._release(inner, pure)(deps),
-            )(IOResult.from_value(inner_value))
-        return RequiresContextIOResult(factory)
-
-    async def _future_pipeline(self, inner_value):
-        used = self._use(inner_value)
-        inner_result = await used._inner_value  # noqa: WPS437
-        await self._release(inner_value, inner_result)
-        return FutureResult.from_result(inner_result)
-
-    def _reader_future_result_pipeline(self, inner_value):
-        def factory(deps):
-            return _managed(
-                lambda inner: self._use(inner)(deps),
-                lambda inner, pure: self._release(inner, pure)(deps),
-            )(FutureResult.from_value(inner_value))
-        return RequiresContextFutureResult(factory)
+def _managed(use, release):
```

Comment:
It will be great if you add `type hints` here, it'll help to fix #472

Reply:
Yes, it is done already! I have posted the updated version in our chat.